src/message_bar.rs

```diff
@@ -0,0 +1,476 @@
+            lines.truncate(max_lines);
+            if TRUNCATED_MESSAGE.len() <= num_cols {
+                if let Some(line) = lines.iter_mut().last() {
+                    *line = Self::pad_text(TRUNCATED_MESSAGE.into(), num_cols);
+                }
+            }
+        }
+
+        // Append close button to first line
+        if CLOSE_BUTTON_TEXT.len() <= num_cols {
+            if let Some(line) = lines.get_mut(0) {
+                line.truncate(num_cols - CLOSE_BUTTON_TEXT.len());
+                line.push_str(CLOSE_BUTTON_TEXT);
+            }
+        }
+
+        lines
+    }
+
+    /// Message color
+    #[inline]
+    pub fn color(&self) -> Rgb {
+        self.color
+    }
+
+    /// Message topic
+    #[inline]
+    pub fn topic(&self) -> Option<&String> {
+        self.topic.as_ref()
+    }
+
+    /// Update the message topic
+    #[inline]
+    pub fn set_topic(&mut self, topic: String) {
+        self.topic = Some(topic);
+    }
+
+    /// Right-pad text to fit a specific number of columns
+    #[inline]
+    fn pad_text(mut text: String, num_cols: usize) -> String {
+        let padding_len = num_cols.saturating_sub(text.len());
+        text.extend(vec![' '; padding_len]);
+        text
+    }
+}
+
+/// Storage for message bar
+#[derive(Debug)]
+pub struct MessageBar {
+    current: Option<Message>,
+    messages: Receiver<Message>,
+    tx: Sender<Message>,
+}
+
+impl MessageBar {
+    /// Create new message bar
+    pub fn new() -> MessageBar {
+        let (tx, messages) = crossbeam_channel::unbounded();
+        MessageBar {
+            current: None,
+            messages,
+            tx,
+        }
+    }
+
+    /// Check if there are any messages queued
+    #[inline]
+    pub fn is_empty(&self) -> bool {
+        self.current.is_none()
+    }
+
+    /// Current message
+    #[inline]
+    pub fn message(&mut self) -> Option<Message> {
+        if let Some(current) = &self.current {
+            Some(current.clone())
+        } else {
+            self.current = self.messages.try_recv().ok();
+            self.current.clone()
+        }
+    }
+
+    /// Channel for adding new messages
+    #[inline]
+    pub fn tx(&self) -> Sender<Message> {
+        self.tx.clone()
+    }
+
+    /// Remove the currently visible message
+    #[inline]
+    pub fn pop(&mut self) -> Option<Message> {
+        std::mem::replace(&mut self.current, self.messages.try_recv().ok())
+    }
+
+    /// Remove all messages with a specific topic
+    #[inline]
+    pub fn remove_topic(&mut self, topic: String) {
+        // Remove the currently active message
+        while self.current.as_ref().and_then(|m| m.topic()) == Some(&topic) {
+            self.pop();
+        }
+
+        // Filter messages currently pending
+        for msg in self
+            .messages
+            .try_iter()
+            .take(self.messages.len())
+            .filter(|m| m.topic() != Some(&topic))
+        {
+            let _ = self.tx.send(msg);
```

Comment:
I feel like this would be mostly confusing, since it doesn't actually do that.

It just filters out all messages, however it does *not* display, or even load the next message. That is only done once the `message()` function is called.